packages/vscode-extension/src/commands/CreateFile.ts

```diff
@@ -0,0 +1,209 @@
+					window.showTextDocument(createdFile);
+				},
+				(error: any) => {
+					this.messageService.showCommandError(error);
+				}
+			);
+		} else {
+			this.messageService.showCommandError();
+		}
+	}
+
+	private async getCommandArgs(): Promise<{ [key: string]: string | string[] } | undefined> {
+		const args: { [key: string]: string | string[] } = {
+			project: this.getProjectFolderPath(),
+		};
+
+		const selectedScriptType = await this.promptScriptTypeQuestion();
+		if (selectedScriptType === undefined) {
+			return;
+		} else {
+			args.type = selectedScriptType.id;
+		}
+
+		const selectedModules = await this.promptAddModulesQuestion();
+		if (selectedModules === undefined) {
+			return;
+		} else if (Array.isArray(selectedModules) && selectedModules.length > 0) {
+			args.module = selectedModules;
+		}
+
+		const selectedFolder = await this.promptFolderSelection();
+		if (selectedFolder === undefined) {
+			return;
+		}
+		const fileName = await this.promptFileNameInputBox(selectedFolder);
+		if (fileName === undefined) {
+			return;
+		}
+
+		args.path = path.join(selectedFolder, fileName);
+
+		return args;
+	}
+
+	private promptScriptTypeQuestion(): Thenable<SuiteScriptTypeItem | undefined> {
+		return window.showQuickPick(
+			SUITESCRIPT_TYPES.map((el) => <SuiteScriptTypeItem>{ label: el.name, id: el.id, name: el.name }),
+			{
+				placeHolder: this.translationService.getMessage(CREATE_FILE.QUESTIONS.CHOOSE_SUITESCRIPT_TYPE),
+				canPickMany: false,
+			}
+		);
+	}
+
+	private async promptAddModulesQuestion(): Promise<string[] | undefined> {
+		return window.showQuickPick(
+			SUITESCRIPT_MODULES.map((module) => module.id),
+			{
+				placeHolder: this.translationService.getMessage(CREATE_FILE.QUESTIONS.SELECT_SUITESCRIPT_MODULES),
+				canPickMany: true,
+			}
+		);
+	}
+
+	private async promptFolderSelection(): Promise<string | undefined> {
+		const folderChoices = this.getFolderChoices();
+
+		let fileToCheck = this.activeFile;
+
+		// action orignated from context menu
+		if (this.isFileSelected && fileToCheck && fs.existsSync(fileToCheck)) {
+			const fileCabinetService = new FileCabinetService(path.join(this.getProjectFolderPath(), ApplicationConstants.FOLDERS.FILE_CABINET));
+			if (!fs.lstatSync(fileToCheck).isDirectory()) {
+				fileToCheck = path.dirname(fileToCheck);
+			}
+			// filter folderChoices by the selected folder in the treeview
+			const filteredFolderChoices = folderChoices.filter((folder) =>
+				folder.startsWith(fileCabinetService.getFileCabinetRelativePath(fileToCheck))
+			);
+			// Autoselect folder when no subfolders in the tree
+			if (filteredFolderChoices.length === 1) {
+				return filteredFolderChoices[0];
+			}
+			return window.showQuickPick(filteredFolderChoices, {
+				placeHolder: this.translationService.getMessage(CREATE_FILE.QUESTIONS.SELECT_FOLDER),
+				canPickMany: false,
+			});
+		}
+
+		// action not originated from context menu
+		return window.showQuickPick(folderChoices, {
```

Comment:
This lines are repeated from previous lines 151-154

Reply:
They are not the same, one is using `filteredFolderChoices` and the other one `folderChoices`.

Reply:
you could overwrite folderChoices instead of creating new variable filteredFolderChoices.
But anyways, if you think is more readable this way, it's ok